### PR TITLE
refactor(components): migrate to Angular's built-in control flow

### PR DESCRIPTION
**Pull request by Marcarrian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-toolkit/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added
- [ ] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Fix
- [ ] Feature
- [ ] Documentation
- [x] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, removal of deprecations, etc)
- [ ] Deps (changes related to updating dependencies)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/components/src/app/sci-filter-field-page/sci-filter-field-page.component.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@
  */
 import {Component} from '@angular/core';
 import {SciFilterFieldComponent} from '@scion/components.internal/filter-field';
-import {NgIf} from '@angular/common';
+
```
Comment (k-genov) [inline apps/components/src/app/sci-filter-field-page/sci-filter-field-page.component.ts:12]: performance line?

File: apps/components/src/app/sci-list-page/sci-list-page.component.html [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 <sci-list (filter)="onFilter($event)" filterPosition="top">
-  <ng-container *ngFor="let listItem of items$ | async">
+  @for (listItem of items$ | async; track listItem) {
     <!-- list item -->
     <ng-template sciListItem [actions]="[flag_action]">
-      <mark *ngIf="isFlagged(listItem)">
+      @if (isFlagged(listItem)) {
+        <mark> {{listItem}}</mark>
+      }
+      @if (!isFlagged(listItem)) {
```
Comment (k-genov) [inline apps/components/src/app/sci-list-page/sci-list-page.component.html:10]: else?

File: apps/components/src/app/sci-list-page/sci-list-page.component.html [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
     </ng-template>
-
     <!-- action -->
     <ng-template #flag_action>
-      <button *ngIf="!isFlagged(listItem)" (click)="onFlag(listItem)" sciMaterialIcon>check_box_outline_blank</button>
-      <button *ngIf="isFlagged(listItem)" (click)="onUnflag(listItem)" sciMaterialIcon>check_box</button>
+      @if (!isFlagged(listItem)) {
+        <button (click)="onFlag(listItem)" sciMaterialIcon>check_box_outline_blank</button>
+      }
+      @if (isFlagged(listItem)) {
```
Comment (k-genov) [inline apps/components/src/app/sci-list-page/sci-list-page.component.html:19]: else?

File: projects/scion/components/sashbox/src/sashbox.component.html [hunk 4]
```diff
@@ -8,13 +8,13 @@
...
-  </sci-splitter>
-</ng-container>
+  @if (!last) {
+    <sci-splitter sciElementRef #sciElementRef="sciElementRef"
+                  [orientation]="isRowDirection ? 'vertical' : 'horizontal'"
+                  (start)="onSashStart()"
+                  (move)="onSash(sciElementRef.nativeElement, i, $event)"
+                  (end)="onSashEnd()"
+                  (reset)="onSashReset(i)">
+    </sci-splitter>
```
Comment (k-genov) [inline projects/scion/components/sashbox/src/sashbox.component.html:18]: auto-close?

File: docs/site/tools/sashbox.md [hunk 5]
```diff
@@ -111,7 +111,7 @@ Sash content modeled in the `<ng-template>` is added to a CSS grid container wit
 <details>
   <summary><strong>Sash Template</strong></summary>
   
-  A sash is added to the sashbox in the form of a `<ng-template>` decorated with the `sciSash` directive. You can control its size by setting a `size` and/or `minSize`. To hide a sash, for example if using the sash as side panel, add a `*ngIf` to the sash `<ng-template>`.
+  A sash is added to the sashbox in the form of a `<ng-template>` decorated with the `sciSash` directive. You can control its size by setting a `size` and/or `minSize`. To hide a sash, for example if using the sash as side panel, add a `@if` to the sash `<ng-template>`.
```
Comment (k-genov) [inline docs/site/tools/sashbox.md:114]: add a `@if` to the sash => place the sash in an `@if` block

File: projects/scion/components.internal/list/src/list.component.html [hunk 6]
```diff
@@ -1,19 +1,23 @@
...
+  }
 </sci-viewport>
 
-<ng-container *ngIf="filterPosition === 'bottom'; then filter_template"></ng-container>
+@if (filterPosition === 'bottom') {
+  <ng-container *ngTemplateOutlet="filter_template"></ng-container>
+}
 
-<!-- filter field if positioned at the bottom -->
 <ng-template #filter_template>
```
Comment (k-genov) [inline projects/scion/components.internal/list/src/list.component.html:21]: Do we still need this ng-template?

Comment (Marcarrian) [reply projects/scion/components.internal/list/src/list.component.html:21]: Yes, otherwise the code is duplicated

**Review by k-genov — APPROVED** [review]
lgtm

**Pull request merged**